### ISSUE TITLE
DOS Risk in Regular Sync Due to Post-DB KZG Checks

### ISSUE THREAD

**Issue by terencechain** [prysmaticlabs/prysm#13055]
#### Summary:
Performing KZG checks after DB writes in regular sync exposes a DOS risk. The proposer must be willing sacrifice the block to exploit it.

#### Proposed Solution:
- Save blobs to cache first.
- Perform KZG checks on cached blobs.
- Move blobs to DB only after passing checks.

**Comment by terencechain** [comment]
No longer relevant after new blob side car format
